source/Fleet.h

```diff
@@ -61,9 +69,12 @@ class Fleet {
 	const Government *GetGovernment() const;
 
 	// Choose a fleet to be created during flight, and have it enter the system via jump or planetary departure.
-	void Enter(const System &system, std::list<std::shared_ptr<Ship>> &ships, const Planet *planet = nullptr) const;
+	void Enter(const System &system, std::list<std::shared_ptr<Ship>> &ships, std::vector<FleetHolder> &fleets,
```

Comment:
Why does the Fleet need the internal state of the Engine? If you're putting a spawned fleet into a container, can't you just return the container? (Or a shared_ptr to it?)

Reply:
done, somehow, great overhaul